lifter/Semantics.cpp

```diff
@@ -259,33 +259,29 @@ void lifterClass::branchHelper(Value* condition, const string& instname,
   // cout << "pathInfo:" << pathInfo << " dest: " << destination  <<
   // "\n";
 }
-
-void lifterClass::lift_movsb() {
+void lifterClass::lift_movsX() {
+  // Get the size based on the operand
+  int size = operands[1].size;
 
   // DEST := SRC;
-  // [esi] = [edi]
+  // Parameterize size and direction
   // sign = DF (-1/+1)
-  // incdecv = size*sign (sb means size is 1)
-  // esi += incdecv
-  // edi += incdecv
-  //
-
-  // Value* SRCptrvalue =
-  // GetOperandValue(operands[0],operands[0].size);
-
-  Value* DSTptrvalue = GetOperandValue(operands[1], operands[1].size);
+  // incdecv = size*sign
 
+  Value* DSTptrvalue = GetOperandValue(operands[1], size);
   SetOperandValue(operands[0], DSTptrvalue);
 
   bool isREP = (instruction.attributes & ZYDIS_ATTRIB_HAS_REP) != 0;
 
   Value* DF = getFlag(FLAG_DF);
+  auto byteSize = ConstantInt::get(DF->getType(), size);
   auto one = ConstantInt::get(DF->getType(), 1);
-  // sign = (x*(x+1)) - 1
-  // v = sign * bytesize ; bytesize is 1
 
-  Value* Direction =
-      createSubFolder(createMulFolder(DF, createAddFolder(DF, one)), one);
+  // sign = (DF*(DF+1)) - 1
+  // v = sign * byteSize
+  Value* Direction = createMulFolder(
+      createSubFolder(createMulFolder(DF, createAddFolder(DF, one)), one),
+      byteSize);
```

Comment:
Now that I look back, we can just use a select here, 
if DF == 0, direction is -1, 
if DF == 1, direction is  1

Reply:
> Now that I look back, we can just use a select here, if DF == 0, direction is -1, if DF == 1, direction is 1

I tried with 
```cpp
  Value* Direction =
      createSelectFolder(DF, ConstantInt::get(DF->getType(), 1),
        ConstantInt::get(DF->getType(), -1));
```

But it seems does something wrong.

Reply:
Uhhh, I just realized, why are we doing `DF->getType()` ? Also should be 
```
 Value* Direction =
      createSelectFolder(DF, ConstantInt::get(__directionsize__, 1*bytesize),
        ConstantInt::get(__directionsize__, -1*bytesize));
        ```

Reply:
That worked.